admin/admin-filters-term.php

```diff
@@ -118,8 +118,15 @@ public function add_term_form() {
 
 		$dropdown = new PLL_Walker_Dropdown();
 
+		$available_languages = $this->model->languages->filter( 'translator' )->get_list();
+
+		// Ensure that the selected language is in the filtered list to display the flag correctly on page load.
+		if ( $lang && ! in_array( $lang, $available_languages, true ) ) {
+			$lang = reset( $available_languages ) ?: $lang;
```

Comment:
I don't think that selecting another language makes sense: this case should never happen because the process will die before that, thanks to https://github.com/polylang/polylang-pro/pull/2744. So we can just bail out IMHO.

```suggestion
			return;
```